src/App.tsx

```diff
@@ -1,61 +1,130 @@
+import { FormEvent, useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const findUserById = (userId: number) => {
+  return usersFromServer.find((user) => (
+    user.id === userId
+  )) || null;
+};
+
+const todos = todosFromServer.map((todo) => {
+  return {
+    ...todo,
+    user: findUserById(todo.userId),
+  };
+});
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userName, setUserName] = useState('');
+  const [visibleTodos, setTodos] = useState(todos);
+  const [titleValid, checkTitleValid] = useState(true);
+  const [userNameValid, checkUserNameValid] = useState(true);
+
+  const handleTitle = (titleValue: string) => {
+    checkTitleValid(true);
+    setTitle(titleValue);
+  };
+
+  const handleUser = (chosenValue: string) => {
+    checkUserNameValid(true);
+    setUserName(chosenValue);
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      checkTitleValid(false);
+    }
+
+    if (!userName) {
+      checkUserNameValid(false);
+    }
+
+    const chosenUser = usersFromServer
+      .find((user) => (user.name === userName)) || null;
+
+    if (title && userName) {
```

Comment:
if you don't have a title or user - make a return
so, after that remove this if